src/Microsoft.AspNetCore.Cors/Infrastructure/UriExtensions.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Cors.Infrastructure
+{
+    internal static class UriExtensions
+    {
+        public static bool IsSubdomainOf(this Uri subdomain, Uri domain)
```

Comment:
We generally avoid extension methods in cases like this - we prefer just a plain old static helper method.